libsolidity/ast/AST.cpp

```diff
@@ -38,6 +40,20 @@ using namespace std;
 using namespace solidity;
 using namespace solidity::frontend;
 
+namespace
+{
+TryCatchClause const*
+checkForClause(vector<ASTPointer<TryCatchClause>> const& clausesList, optional<string> stringPred = {})
+{
+	for (auto const& clause: ranges::views::tail(clausesList))
+	{
+		if ((stringPred) ? clause->errorName() == stringPred : std::empty(clause->errorName()))
```

Comment:
```suggestion
		if (stringPred.has_value() ? clause->errorName() == stringPred : clause->errorName().empty())
```

Reply:
This is an interesting suggestion and I liked discovering this suggested change. 

_(I personally wonder,)_ the challenge of understanding a codebase is "how much is _standardised_?". As your mentioned[ another comment, clear intent is a positive](https://github.com/ethereum/solidity/pull/13409#discussion_r949231577), and `std::empty` implies the datatype's `.empty()` follows `std::` standards.

However, for this project `solidity`, the `std::` namespace is name-masked, and having the code's text as `empty(clause->errorName())` is very awkward in understanding its intent... 

So, `clause->errorName().empty()`, is the only option one can take.

In no way saying anything negative about the suggestion, just more noting that I agree with the suggestion and that the suggestion is "interesting". :relaxed:

Reply:
I mean, `empty(clause->errorName())` would be fine. It's just that `clause->errorName().empty()` is how we usually do it and I find it cleaner but it's subjective. I wouldn't really reject the PR if you insisted on the former but just wanted to show how I'd do it.